### PR TITLE
Added exception handling to critical aws api calls and aws account resource retention policies

### PR DESCRIPTION
**Pull request by shawngerrard**
I've created an access key for the newly created iam user to be able to make api calls when policies permit.

I've also defined an admin role in the master account. The eventual intention for the role is that it will be assumed for elevated EC2 privileges in the asterion member accounts.

### CODE REVIEW

File: org-asterion/__main__.py [hunk 1]
```diff
@@ -87,17 +87,27 @@ def org_exists(self):
...
+new_user_access_key = aws.iam.AccessKey(
+    "asterion-infra-aws-user-access-key",
+    user=new_user.name
+)
+
 # Export password for the user
 export("New user password", new_user_login.password)
 
+# Export access key for the user
+export("Encrypted secret access key", new_user_access_key.encrypted_secret)
```
Comment (jmhbnz) [inline org-asterion/__main__.py]: Curious, are these values random / dynamically generated when you create the key on line 91?

**Review by jmhbnz — APPROVED** [review]

**Comment by jmhbnz** [comment]
Looks good to me, nice increment 👍🏻

File: org-asterion/__main__.py (hunk 1 above)
Comment (shawngerrard) [reply org-asterion/__main__.py]: They are if there's a diff on the login profile resource - you can force a password reset using this method. 

**Note**: I'm removing the secret access key generation as part of this process as I feel administrators should do this themselves.

I'm also removing the role from here as this should reside in the eventual `org-asterion-dev` stack. This means I'll be updating the parameters on this pull request and will request re-review shortly.

**Comment by shawngerrard** [comment]
> Looks good to me, nice increment 👍🏻

@jmhbnz - I've updated the pull request title to reflect the scope of changes here. Apologies for the last min scope change. The user role doesn't belong in the `org-asterion` stack, and I think it's more secure and better suited to leave CLI access key generation out of the scope of this codebase as long as the iam users can log in to the aws console.

Also, I've included aws account resource retention in this update as it's important to mark these resources with this option up-front so that the pulumi stack can be destroyed in future and the accounts can be managed manually with the provider (or the code updated so that these resources can be reclaimed as per the retention pattern in issue #22).

Have requested a re-review of the changes.

**Review by jmhbnz — APPROVED** [review]

**Pull request merged**